src/server/services/BruincastServices.js

```diff
@@ -95,68 +95,38 @@ class BruincastServices {
     return castsByCourses;
   }
 
-  static formatCastListings(castEntries) {
-    const formattedArray = [];
-    let currentShortname = '';
-    let currentListings = [];
-    for (const [i, entry] of castEntries.entries()) {
-      const entryClassShortname = entry.classShortname;
-
-      let filetype = 'Audio/Video';
-      let file = '';
-      if (entry.video === '' && entry.audio === '') {
-        filetype = 'None';
-      } else if (entry.video === '') {
-        filetype = 'Audio';
-        file = entry.audio;
-      } else if (entry.audio === '') {
-        filetype = 'Video';
-        file = entry.video;
-      } else {
-        file = `${entry.audio}, ${entry.video}`;
-      }
-
-      const formattedEntry = {
-        term: entry.term,
-        classID: entry.classID,
-        week: entry.week,
-        date: entry.date,
-        title: entry.title,
-        comments: dompurify.sanitize(entry.comments),
-        type: filetype,
-        filename: file,
-      };
-
-      if (i === 0) {
-        currentShortname = entryClassShortname;
-      }
-
-      if (entryClassShortname !== currentShortname) {
-        formattedArray.push({
-          courseShortname: currentShortname,
-          courseListings: currentListings,
-        });
-        currentListings = [];
-        currentShortname = entryClassShortname;
-      }
-
-      currentListings.push(formattedEntry);
+  static formatTermCasts(media) {
+    for (const courseMedia of media) {
+      for (const entry of courseMedia.listings) {
```

Comment:
Oh, Rex told me that for each bruincast entry in the database, there can be multiple audios and videos in the form of comma-separated strings, something like 'xxx.mp4,yyy.mp4'. You might want to do an entry.video.split(',') and then a loop.

Reply:
Will it mess up how your admin listing works?

Reply:
Oh ok, I didn't realize that there could be multiple files, but it shouldn't affect the listings because each listing corresponds to a record in the db, and the listings concatenate the audio and video strings into one string